js/ui/metric-legend.js

```diff
@@ -21,5 +25,6 @@ export const MetricLegend = {
       </div>
       <span v-html="metric.htmlName"></span>
       <img v-if="metric.name" src="img/icons/pencil.svg" />
```

Comment:
For consitency reasons, the edit action should also only trigger on pressing the pencil, rather than the full legend element.